pytorch_lightning/utilities/auto_restart.py

```diff
@@ -351,6 +356,11 @@ def __next__(self) -> Dict[str, Any]:
         return next(self.iter_data)
 
 
+def is_obj_stateful(obj: Any) -> bool:
+    """In order to be stateful, an object should implement a ``state_dict`` and ``load_state_dict`` method."""
+    return getattr(obj, "state_dict", None) is not None and getattr(obj, "load_state_dict", None) is not None
```

Comment:
use hasattr? 
it may need a callable check?

Reply:
Done !